pkg/util/cloudproviders/cloudproviders.go

```diff
@@ -49,6 +50,15 @@ func DetectCloudProvider(ctx context.Context) {
 		if cloudDetector.callback(ctx) {
 			inventories.SetAgentMetadata(inventories.HostCloudProvider, cloudDetector.name)
 			log.Infof("Cloud provider %s detected", cloudDetector.name)
+
+			// fetch the account ID for this cloud provider
+			if cloudDetector.accountIdCallback != nil {
```

Comment:
Would it make sens to have a feature flag around this ?

Reply:
I can add a feature flag, but I would like it to be enabled by default if possible otherwise it kind of defeats the purpose. WDYT ?